spec/abci/abci.md

```diff
@@ -45,33 +45,34 @@ More details on managing state across connections can be found in the section on
 
 ## Errors
 
-Some methods (`Echo, Info, InitChain, BeginBlock, EndBlock, Commit`),
-don't return errors because an error would indicate a critical failure
-in the application and there's nothing Tendermint can do. The problem
-should be addressed and both Tendermint and the application restarted.
+The `Query`, `CheckTx` and `DeliverTx` methods include a `Code` field in their `Response*`.
+This field is meant to contain an application-specific response code.
+A response code of `0` indicates no error.  Any other response code
+indicates to Tendermint that an error occurred.
 
-All other methods (`Query, CheckTx, DeliverTx`) return an
-application-specific response `Code uint32`, where only `0` is reserved
-for `OK`.
-
-Finally, `Query`, `CheckTx`, and `DeliverTx` include a `Codespace string`, whose
-intended use is to disambiguate `Code` values returned by different domains of the
+These methods also return a `Codespace` string to Tendermint. This field is
+used to disambiguate `Code` values returned by different domains of the
 application. The `Codespace` is a namespace for the `Code`.
 
+The `Echo`, `Info`, `InitChain`, `BeginBlock`, `EndBlock`, `Commit` methods
+do not return errors. An error in any of these methods represents a critical
+issue that Tendermint has no reasonable way to handle. If there is an error in one
+of these methods, the application must crash to ensure that the error is safely
+handled by an operator.
+
 ## Events
 
-Some methods (`CheckTx, BeginBlock, DeliverTx, EndBlock`)
-include an `Events` field in their `Response*`. Each event contains a type and a
-list of attributes, which are key-value pairs denoting something about what happened
-during the method's execution.
+The `CheckTx`, `BeginBlock`, `DeliverTx`, `EndBlock` methods include an `Events`
+field in their `Response*`. Each event contains a `type` and a list of `EventAttribute`s,
+which are key-value pairs denoting something about what happened during the method's execution.
 
-Events can be used to index transactions and blocks according to what happened
+`Event` values can be used to index transactions and blocks according to what happened
```

Comment:
![events](https://user-images.githubusercontent.com/27455974/131698102-0246eb85-a171-4e9c-84e8-2ad2fd1198bf.jpg)

I find adding a schematic to the descriptions will help with the more visual learners

Reply:
I think it's not a bad call, but I'd prefer to may do a bit more polishing if adding an image. My goal with this PR was to spot-fix a few issues or ambiguities I noticed in the spec as it related to issue #316.